### DISCUSSION TITLE
Repeater can get messy, suggestion for enhancements

### DISCUSSION THREAD

**Discussion by sandersjj** [filamentphp/filament#10599] [General]
When utilizing the Repeater component for organizing order items within an order structure, managing a large number of order lines can be challenging, impacting the overall order overview. I've prepared a screenshot with UI suggestions aimed at enhancing this experience.

I realize that using the repeater for order lines isn't the only use case for repeater but I have the feeling it is quite a common use case. 

So please take a look at the screenshot and maybe I will be able to create a PR or someone else in the community might have a different suggestion. 

Thanks

![image](https://github.com/filamentphp/filament/assets/176772/1e92a15e-3421-40f3-bfc1-e1e9be904292)

**Comment by zepfietje** [comment]
This is actually an idea I coined when we introduced the [simple repeater](https://filamentphp.com/docs/3.x/forms/fields/repeater#simple-repeaters-with-one-field).
The issue is though that there's multiple labels with multiple fields.

The [table repeater](https://github.com/awcodes/filament-table-repeater) package by @awcodes might actually be what you're looking for.

However, I still do think a simple repeater with multiple fields per line has a place in core.

CC @danharrin (please contact me via Discord if you think it's worth discussing, I won't be notified here)

**Comment by sandersjj** [comment]
![Screenshot 2024-01-02 at 20 50 07](https://github.com/filamentphp/filament/assets/176772/11d84501-357b-4583-a2db-347c618bd4b4)
This gives indeed the wanted experience. 

Thanks @zepfietje for the tip. And awesome work @awcodes 

I will close this for now as this suffices my use case.
